### PR TITLE
Using federated auth for ACS

### PR DESCRIPTION
**Pull request by rajuanitha88**
Changes:
In order to move away from Secret based authentication, we want to adopt FIC.
For the same azure-sdk team has added the support and we need to use service connection for our live test pipelines to access our test resources.

### CODE REVIEW

File: sdk/communication/azure-communication-identity/tests.yml [hunk 1]
```diff
@@ -5,18 +5,13 @@ extends:
...
+    Clouds: 'Public'
+    UseFederatedAuth: true
     CloudConfig:
       Public:
+        ServiceConnection: azure-sdk-tests
+        SubscriptionConfigurationFilePaths:
+          - eng/common/TestResources/sub-config/AzurePublicMsft.json
         SubscriptionConfigurations:
-          - $(sub-config-azure-cloud-test-resources)
           - $(sub-config-communication-services-cloud-test-resources-common)
```
Comment (mjafferi-msft) [inline sdk/communication/azure-communication-identity/tests.yml:16]: Can we use the new secret we created just for CTE here as well and remove the other ones ?

**Review by mjafferi-msft — APPROVED** [review]
Just 1 small comment to check if we can use CTE secret here as well to maintain consistency across all the platforms, other looks good to me.

**Review by maximrytych-ms — APPROVED** [review]

**Review by mikehang-msft — APPROVED** [review]

File: sdk/communication/azure-communication-identity/tests.yml (hunk 1 above)
Comment (rajuanitha88) [reply sdk/communication/azure-communication-identity/tests.yml:16]: Made the names consistent across all SDKs. Hence removed CTE and reusing the old KV.

**Review by minwoolee-msft — APPROVED** [review]

File: sdk/communication/azure-communication-identity/tests.yml (hunk 1 above)
Comment (weshaggard) [inline sdk/communication/azure-communication-identity/tests.yml:16]: Are there still unique configurations in these configs that are still needed? I see some others like callautomation which don't need these configurations.

Comment (rajuanitha88) [reply sdk/communication/azure-communication-identity/tests.yml:16]: The others are extending a common template: sdk/communication/communication-tests-template.yml where UseFederatedAuth is defaulted to false.

I had to override the flag as true in each of the test yml

Comment (weshaggard) [reply sdk/communication/azure-communication-identity/tests.yml:16]: Is there a reason to not flip that shared template default to true?

Comment (rajuanitha88) [reply sdk/communication/azure-communication-identity/tests.yml:16]: Just to be on the safe side of unneccessarily using federated auth.

**Pull request merged**